ci: Use all available CPUs for functional tests in "Win64 native" task

hebasto opened the pull request:
On the [master](https://cirrus-ci.com/task/5422842484359168) branch:
![Screenshot from 2022-10-12 09-45-58](https://user-images.githubusercontent.com/32963518/195296883-3852ea09-7345-4166-b855-7704dcd87202.png)

This [PR](https://cirrus-ci.com/task/6392972617973760) branch:
![Screenshot from 2022-10-12 11-11-15](https://user-images.githubusercontent.com/32963518/195315902-f667874a-8aeb-4f2f-bdc3-5ba432ae9353.png)


Also consider "CPU Usage" charts provided by CI.

Overlooked in cda62657e95a90a5fd61ba43e2acbd407e3a4135 (bitcoin/bitcoin#25929).

hebasto committed:
ci: Use all available CPUs for functional tests in "Win64 native" task

aureleoules reviewed (approved):
ACK 6fbd173d8a4519967d75d2707b2285d62faa4424
CI error seems unrelated

MarcoFalke commented on .cirrus.yml:
```diff
@@ -184,7 +184,7 @@ task:
     - netsh int ipv6 set dynamicport tcp start=1025 num=64511
     # Exclude feature_dbcrash for now due to timeout
-    - python test\functional\test_runner.py --nocleanup --ci --quiet --combinedlogslen=4000 --jobs=4 --timeout-factor=8 --extended --exclude feature_dbcrash
+    - python test\functional\test_runner.py --nocleanup --ci --quiet --combinedlogslen=4000 --jobs=6 --timeout-factor=8 --extended --exclude feature_dbcrash
```
```suggestion
    - python test\functional\test_runner.py --nocleanup --ci --quiet --combinedlogslen=4000 --jobs=12 --timeout-factor=8 --extended --exclude feature_dbcrash
```

There should be no risk in saturating the CPU, given enough memory

hebasto replied:
Implemented in https://github.com/bitcoin/bitcoin/pull/26297#issuecomment-1275931299.

hebasto commented:
Updated 6fbd173d8a4519967d75d2707b2285d62faa4424 -> dc869c6a77c4542d4d9dcefdec577d1876e51061 ([pr26297.01](https://github.com/hebasto/bitcoin/commits/pr26297.01) -> [pr26297.02](https://github.com/hebasto/bitcoin/commits/pr26297.02), [diff](https://github.com/hebasto/bitcoin/compare/pr26297.01..pr26297.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/26297#discussion_r993220691)

Let CI task finish to see the actual effect of a possible over-subscription.

hebasto commented:
> Looks like this makes the intermittent issues more common

Indeed. Reverted back to 6fbd173d8a4519967d75d2707b2285d62faa4424 ([pr26297.01](https://github.com/hebasto/bitcoin/commits/pr26297.01)), which was already [ACKed](https://github.com/bitcoin/bitcoin/pull/26297#pullrequestreview-1138724890) by @aureleoules.

jarolrod reviewed (approved):
ACK 6fbd173d8a4519967d75d2707b2285d62faa4424

Lgtm

shaavan reviewed (approved):
ACK 6fbd173d8a4519967d75d2707b2285d62faa4424

Makes sense to increase number of jobs equal to number of CPUs.

aureleoules commented:
The merge commit description included hebasto's comment even tough it's not a ACK @MarcoFalke, probably not that big of a deal though.
 
> ACKs for top commit:
  hebasto:
    Indeed. Reverted back to https://github.com/bitcoin/bitcoin/commit/6fbd173d8a4519967d75d2707b2285d62faa4424 ([pr26297.01](https://github.com/hebasto/bitcoin/commits/pr26297.01)), which was already [ACKed](https://github.com/bitcoin/bitcoin/pull/26297#pullrequestreview-1138724890) by @aureleoules.
  aureleoules:
    ACK https://github.com/bitcoin/bitcoin/commit/6fbd173d8a4519967d75d2707b2285d62faa4424
  jarolrod:
    ACK https://github.com/bitcoin/bitcoin/commit/6fbd173d8a4519967d75d2707b2285d62faa4424
  shaavan:
    ACK https://github.com/bitcoin/bitcoin/commit/6fbd173d8a4519967d75d2707b2285d62faa4424

MarcoFalke commented:
I think we should fix the intermittent issues and then set the jobs to 12

hebasto commented:
> and then set the jobs to 12

Why? CPUs are used quite effectively [now](https://cirrus-ci.com/task/5130552108908544):

![image](https://user-images.githubusercontent.com/32963518/196999773-2ea97232-579f-4e28-ad11-c42e53621a58.png)

MarcoFalke commented:
> Why

So that intermittent issues are happening more consistently and can then hopefully be found and fixed before they make it into the master branch
